[Lint]: update lint packages, fix lint errors

hzoo opened the pull request:
@cesarandreu pointed out running eslint on react-native resulted in `t.isReferencedIdentifier is not a function` issues on the babel #linting channel on slack.

- update eslint, babel-eslint, eslint-plugin-react
- fix eslint errors: `Error - t.isReferencedIdentifier is not a function`
- fix lint npm script

I see there's also https://github.com/facebook/react-native/pull/1736 from @ide which would fix it too

cesarandreu commented:
:+1:

cyprusglobe commented:
:+1:

tadeuzagallo commented:
@facebook-github-bot import

hzoo commented:
Maybe I should update this again? (although with eslint becoming 1.0 there probably needs to be some more changes)

```
+    "babel-eslint": "4.0.5",
+    "eslint": "1.1.0",
+    "eslint-plugin-react": "3.2.2"
```

tadeuzagallo commented:
That'd be great :)

frantic commented:
Please rebase :)

hzoo commented:
Rebased - however can update the 3 packages again

```
    "babel-eslint": "4.0.10",
    "eslint": "1.1.0",
    "eslint-plugin-react": "3.2.3"
```
which would require more changes since eslint is 1.0 which I can do as well (in this PR or a new one)

ide commented:
Here's an eslintrc 1.0 I use, which should be quite close to the one that RN ships with. It's not exactly the same but thought it could be helpful: https://gist.github.com/ide/4c9b3de2ae9e952c8b3d

hzoo commented:
K will update to latest versions then - we'll see if I can incorporate some of @ide's config (seems like eslint nicely warns what new rules to use instead of deprecated ones)

hzoo commented:
```
Rule 'no-empty-class' was removed and replaced by: no-empty-character-class
Rule 'global-strict' was removed and replaced by: strict
Rule 'no-extra-strict' was removed and replaced by: strict
Rule 'no-space-before-semi' was removed and replaced by: semi-spacing
Rule 'no-wrap-func' was removed and replaced by: no-extra-parens
```

Didn't bother fixing any errors (getting `209 problems (92 errors, 117 warnings)`)

hzoo commented:
Fixed a merge conflict
